jgrapht-core/src/test/java/org/jgrapht/traverse/LexBreadthFirstIteratorTest.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * (C) Copyright 2018, by Timofey Chudakov and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.traverse;
+
+import org.jgrapht.Graph;
+import org.jgrapht.graph.DefaultEdge;
+import org.jgrapht.graph.DefaultUndirectedGraph;
+import org.jgrapht.graph.Pseudograph;
+import org.junit.Test;
+
+import java.util.HashSet;
+import java.util.NoSuchElementException;
+import java.util.Set;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Tests for the {@link LexBreadthFirstIterator}
+ *
+ * @author Timofey Chudakov
+ */
+public class LexBreadthFirstIteratorTest {
```

Comment:
Need test for events.

Reply:
I implemented edgeTraversed firing. Do I understand it right: an iterator is cross-component if it iterates vertices in one connected component and then begins iterating vertices from another connected component? For example, BreadthFirstIterator isn't cross-component if it is created with vertices it should iterate first, otherwise, it iterates vertices across connected components. Both lexicographical breadth-first search and maximum cardinality search don't have any sense if they iterate vertices not across connected components. That is why they don't have to have an option to specify start vertices. Or I don't get it right and it has other meaning?

Reply:
Yes, your understanding is correct.

Regarding the edge traversals, for the existing iterators such as DFS/BFS, there's a well-defined notion of which edge the traversal used.  Is this actually the case for MCS/LexBFS?  If it is, then you should only be firing for a specific edge, not all outgoing.  If it isn't, then you shouldn't be firing for edges at all (see TopologicalIterator).

Reply:
Resting upon articles I've read there is no such definition. It is possible to define which edge is traversed on the analogy of BFS, but I think it'll be better not to do it. The reason is there are actually two possible definitions: all outgoing edges of a vertex or only those edges, which are incident to the vertices, which haven't been visited.